src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
-
-      // Creates a new measure tooltip with a nice arrow
-      var createMeasureTooltip = function() {
-        var tooltipElement = $document[0].createElement('div');
-        tooltipElement.className = 'ga-draw-measure';
-        var tooltip = new ol.Overlay({
-          element: tooltipElement,
-          offset: [0, -15],
-          positioning: 'bottom-center',
-          stopEvent: true // for copy/paste
+        select.on('select', function(evt) {
+          if (evt.selected[0]) {
+            // Update the position of the popup according to the click event.
+            togglePopup(evt.selected[0], evt.mapBrowserEvent.coordinate);
+          }
         });
-        return tooltip;
-      };
+        select.getFeatures().on('add', function(evt) {
+           // Apply the select style
+          var styles = scope.options.selectStyleFunction(evt.element);
+          evt.element.setStyle(styles);
+          // Show the popup
+          togglePopup(evt.element);
+        });
+        select.getFeatures().on('remove', function(evt) {
+          // Remove the select style
+          var styles = evt.element.getStyle();
+          styles.pop();
+          evt.element.setStyle(styles);
+          // Hide or move the popup to the next selected feature
+          togglePopup(evt.target.item(0));
+        });
+        select.setActive(false);
 
-      var updateHelpTooltip = function(overlay, type, drawStarted, onFirstPoint,
-          onLastPoint) {
-        var helpMsg = $translate.instant('draw_start_' + type);
-        if (drawStarted) {
-          if (type != 'Point') {
-            helpMsg = $translate.instant('draw_next_' + type);
+        var unselectFeature = function(feature) {
+          var selected = select.getFeatures().getArray();
+          for (var i = 0, ii = selected.length; i < ii; i++) {
+            if (selected[i] == feature) {
+              select.getFeatures().remove(feature);
+              break;
+            }
           }
-          if (onLastPoint) {
-            helpMsg = $translate.instant('draw_snap_last_point_' + type);
+        };
+        var onClosePopup = function(feature) {
+          if (select.getFeatures().getLength() > 1) {
+            return;
           }
-          if (onFirstPoint) {
-            helpMsg = $translate.instant('draw_snap_first_point_' + type);
+          unselectFeature(feature);
+        };
+
+        // Add modify interaction
+        // The modify interaction works with features selected by the select
+        // interaction so no need to activate/deactivate it.
+        // Activate/deactivate the select interaction is enough.
+        var modify = new ol.interaction.Modify({
+          features: select.getFeatures(),
+          style: scope.options.selectStyleFunction
+        });
+        modify.on('modifystart', function(evt) {
+          if (evt.mapBrowserPointerEvent.type != 'singleclick') {
+            body.addClass(cssModify);
+            mapDiv.addClass(cssGrabbing);
```

Comment:
The cssModify needs to be apply on all the application to hide the popup when we move the feature.

 And the grab cursor css is applying only on the map div to override the previous map cursor css.  

cssGrab, cssGrabbing and cssPointer are only map's cursor stuff I don't think it's necessary to add it to the body.